docs/book/.gitignore

```
@@ -0,0 +1 @@
+book
```

```suggestion
generated
```
and remove `docs/book/generated/.gitignore` instead, listing all the files manually there is error prone (as is evident by you havin forgotten to ignore the `config.md` :^)

I think I'd in general prefer to just have all the gitignore directives in the main .gitignore file in fact, so let's remove this one alltogether and list the ignored paths in the top level one

---

So I can move everything up to the top level `.gitignore`. A few things, though:

* We probably still want to ignore `docs/book/book` because that's the generated book source.
* I created the `generated` dir to keep things clean, so to keep that directory we need to do something to keep. I can easily just have all the generated files written directly into the `docs/book/src` directory or something, though, so we don't have to deal with having an empty `.gitignore` or `.gitkeep` there. Things seem to break if the directory doesn't exist when the various functions try to write their respective files out.
* I left `config.md` out to match what's currently in master: <https://github.com/rust-lang/rust-analyzer/blob/master/docs/user/generated_config.adoc> is checked in, and is also missing from <https://github.com/rust-lang/rust-analyzer/blob/master/.gitignore>. Not sure if that's an oversight or intentional.

I'll do some cleanup of all of the above and go from there.

---

This is resolved, but there is still a question below about whether the generated configuration file should be ignored.